mesonpy/__init__.py

```diff
@@ -398,73 +398,13 @@ def _is_native(self, file: Union[str, pathlib.Path]) -> bool:
-            stacklevel=2,
-        )
-
-    def _map_from_heuristics(self, origin: pathlib.Path, destination: pathlib.Path) -> Optional[Tuple[str, pathlib.Path]]:
-        """Extracts scheme and relative destination with heuristics based on the
-        origin file and the Meson destination path.
-        """
-        warnings.warn('Using heuristics to map files to wheel, this may result in incorrect locations')
-        sys_paths = mesonpy._introspection.SYSCONFIG_PATHS
-        # Try to map to Debian dist-packages
-        if mesonpy._introspection.DEBIAN_PYTHON:
-            search_path = origin
-            while search_path != search_path.parent:
-                search_path = search_path.parent
-                if search_path.name == 'dist-packages' and search_path.parent.parent.name == 'lib':
-                    calculated_path = origin.relative_to(search_path)
-                    warnings.warn(f'File matched Debian heuristic ({calculated_path}): {origin} ({destination})')
-                    self._warn_unsure_platlib(origin, destination)
-                    return 'platlib', calculated_path
-        # Try to map to the interpreter purelib or platlib
-        for scheme in ('purelib', 'platlib'):
-            # try to match the install path on the system to one of the known schemes
-            scheme_path = pathlib.Path(sys_paths[scheme]).absolute()
-            destdir_scheme_path = self._install_dir / scheme_path.relative_to(scheme_path.anchor)
-            try:
-                wheel_path = pathlib.Path(origin).relative_to(destdir_scheme_path)
-            except ValueError:
-                continue
-            if sys_paths['purelib'] == sys_paths['platlib']:
-                self._warn_unsure_platlib(origin, destination)
-            return 'platlib', wheel_path
-        return None  # no match was found
-
     def _map_from_scheme_map(self, destination: str) -> Optional[Tuple[str, pathlib.Path]]:
-        """Extracts scheme and relative destination from Meson paths.
-
-            Meson destination path -> (wheel scheme, subpath inside the scheme)
-        Eg. {bindir}/foo/bar       -> (scripts, foo/bar)
-        """
-        for scheme, placeholder in [
-            (scheme, placeholder)
-            for scheme, placeholders in self._SCHEME_MAP.items()
-            for placeholder in placeholders
-        ]:  # scheme name, scheme path (see self._SCHEME_MAP)
-            if destination.startswith(placeholder):
-                relative_destination = pathlib.Path(destination).relative_to(placeholder)
-                return scheme, relative_destination
-        return None  # no match was found
+        """Extracts scheme and relative destination from installation paths."""
+        parts = pathlib.Path(destination).parts
+        for scheme, placeholders in self._SCHEME_MAP.items():
+            if parts[0] in placeholders:
+                return scheme, pathlib.Path(os.path.join(*parts[1:]))
```

Comment:
```suggestion
                return scheme, pathlib.Path(*parts[1:])
```

Also, please keep the docstring.

Reply:
Oh, neat. I'm actually quite sure I tried this syntax and that it didn't work. I must have forgot the `*` or something.

Reply:
Yeah, the constructor can't take a sequence as the first arguments, so you need to unpack it.